src/systems/scene_broadcaster/SceneBroadcaster.cc

```diff
@@ -261,16 +261,17 @@ void SceneBroadcaster::Configure(
   auto readHertz = _sdf->Get<int>("dynamic_pose_hertz", 60);
   this->dataPtr->dyPoseHertz = readHertz.first;
 
-  auto stateHerz = _sdf->Get<int>("state_hertz", 60);
+  auto stateHertz = _sdf->Get<double>("state_hertz", 60);
+  if (stateHertz.first < 0.0)
+  {
+    ignerr << "SceneBroadcaster state_hertz must be non-negative\n";
```

Comment:
How about also checking whether it's zero to avoid the division by zero?

And if this check fails, we shouldn't just print an error, but also make sure that the computations below don't end up in undefined behaviour. We could for example set the value to 1 and mention this in the error message.

Reply:
I see, I missed that